### PR TITLE
more accurate estimate of path length and tuned multiplier for two optimisation modes

### PR DESCRIPTION
**Pull request by duanyutong**

### CODE REVIEW

File: toppra/dracula/__init__.py [hunk 1]
```diff
@@ -64,7 +64,13 @@ def _check_waypts(waypts, vlim):
     velocity limit and the waypoints in joint space.
     """
     pair_dist = np.diff(waypts, axis=0)  # (N-1, N_dof)
+    i_sign_flip = (
```
Comment (gavincangan) [inline toppra/dracula/__init__.py]: Please add some comments explaining what this is doing

File: toppra/dracula/__init__.py [hunk 2]
```diff
@@ -162,16 +144,58 @@ def lims_obeyed(self, traj, raise_2nd_order):
...
+                " especially for the beginning and the end of the trajectory. "
+                "Toppra might throw a controllability exception. "
+                "Attempting to optimise trajectory anyway..."
+            )
+        # initial x for toppra's path, essentially normalised time on x axis
+        # rescale by given speed limits.
+        # only applies to ParametrizeSpline.
+        self.path_length_limit = 100 * t_sum  # empirical magic number
+        # t_sum is the minimum time required to visit all given waypoints.
+        # toppra generally needs a smaller number for controllabiility.
```
Comment (gavincangan) [inline toppra/dracula/__init__.py:169]: You mean it needs more time than the minimum time, right?

Comment (duanyutong) [reply toppra/dracula/__init__.py:169]: it needs a smaller path length than the fastest possible `t_sum` for better controllability.

File: toppra/dracula/__init__.py (hunk 1 above)
Comment (duanyutong) [reply toppra/dracula/__init__.py]: added

File: toppra/dracula/__init__.py (hunk 2 above)
Comment (gavincangan) [reply toppra/dracula/__init__.py:169]: @duanyutong Please correct me if I am wrong here: my understanding is that `t_sum` is the minumum time needed by the joint with longest path length to complete its trajectory, assuming instant acceleration/deceleration between 0 and max velocity.

**Pull request merged**